### DISCUSSION TITLE
Offer portable alpha version (Linux)

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1359] [Suggestions]
If not too difficult to achieve, would anyone consider making the alpha versions available as a standalone / portable version similar to how the official releases provide.

Thank You

EDIT:
Strange I thought I filled in the info section when I made the post but it is not there?
Anyways, sorry about that, yes I was looking for the Linux version.
I also just updated the title of this post, add the " (Linux)" suffix.
Could also be helpful for Windows users as well, but as you mentioned Windows may already be easy enough to achieve this.

**Comment by JonnyTech** [comment]
AFAIK they already are, the settings folder has an empty doublecmd.inf file that signifies portable settings folder.

**Comment by alexx2000** [comment]
It is easy to make it portable (Windows snapshots). Simply create empty file `settings\doublecmd.inf`. But I guess you talk about Linux version.

@JonnyTech, snapshots does not have this file. So it can be used to update both portable and non-portable versions.

**Comment by advice2020** [comment]
@alexx2000 
Thank you for your response
Sorry about that I edited my orginal post.
Yes I was referring to the Linux version.

**Comment by alexx2000** [comment]
Linux portable packages are being created from Fedora RPM packages. You can convert [Fedora RPM snapshots](https://download.opensuse.org/repositories/home:/Alexx2000:/doublecmd-svn/Fedora_31/x86_64/) into portable version using [this script](https://raw.githubusercontent.com/doublecmd/doublecmd/master/install/linux/convert-rpm-txz.sh)).

**Comment by advice2020** [comment]
@alexx2000
Thank you for your response, I was not aware of this.
Forgive me but I have some questions about this, because bash script did not appear to work for me.
I extracted the rpm file (in my Downloads folder) which creates a “usr” folder.  I tried running bash script outside this folder as well as inside but does not appear to generate standalone version.  When I run any of the doublecmd files in this usr folder, DC DOES open though without using this bash script which I was surprised (the usr folder contains three folders, unlike the portable version that contains all the folders for DC in one folder)

1.) I am using a Debian based DE, should this script still work?
2.) All the files & folders appear to be the same as the deb version, would their be any conflicts of running this rpm version in a Debian based DE?
3.) This one you do not necessarily need to answer (well you don't need to answer any of these questions).
Is there a reason why the portable version wouldn't be set to generate automatically like all the other versions and be made available on snapshots site?

Thank you again

**Comment by alexx2000** [comment]
1, 2) It does not work because you are using this script in a wrong way
3) It is additional time and work

How to use this script:
1. Install patchelf package `sudo apt install patchelf`
2. [Download](https://github.com/doublecmd/doublecmd/archive/refs/heads/master.zip) and extract Double Commander source code
3. Download RPM file (for example `doublecmd-gtk-1.2.0+svn11052+git5cb86ee-985.1.x86_64.rpm`)
4. Execute terminal program and go to 'install/linux' subdirectory in DC source code directory
5. Execute convert-rpm-txz.sh script: `./convert-rpm-txz.sh /path/to/rpm/doublecmd-gtk-1.2.0+svn11052+git5cb86ee-985.1.x86_64.rpm`
6. Got a portable version in `install/linux/release` subdirectory

**Comment by advice2020** [comment]
@alexx2000
Ahhh thank you for sharing this
I am trying to compare this to a technique Skif_off shared with me for creating portable version.  Just trying to see what is the most efficient way of generating a portable version for testing alphas, just want to make sure I am as up to date as possible so my tests can be as accurate as possible.  Haven't fully tested this method out yet, seems a little more complex.  I think I have an idea though.

1.) / 2.) Okay.... whoa I was way off :)
3.) Thank you for answering this, that is understandable.  I apologize for asking, when I read on snapshot site
"These snapshots are generated automatically and are untested"
I thought these were all somehow just automated to generate alpha versions automatically.  I imagined you had a collection of magical elves that make alpha versions.  I was close though, “Install patchelf” :)

Thank you again for taking the time to answer this and for providing the steps how to achieve this.

Really trying to figure out the most efficient way to do this so I can keep up to date version for testing purposes.
Portable is perfect version for this as it does not require an install like the deb version and the Appimage version has some issues due to its own container format.  Just don't want something that I report to be because of a portable version creation error, and not with DC itself.
